File: src/Modules/Contracts/Core.Contract/DTOs/AdapterEvent/AdapterEventDto.cs
using SharedKernel.Enums;

namespace Core.Contract.DTOs.AdapterEvent;

public sealed record AdapterEventDto(
Guid Guid,
string Name,
string Mac,
int ComponentId,
string Command,
int Tag,
DateTime SendAt,
DateTime ReceivedAt,
string Body,
CommandStatus Status,
string Reason,
string Response,
Vendor Vendor,
Guid LocationGuid,
bool IsActive,
bool IsDefault
);

File: src/Web/src/model/Event/AdapterEventDto.ts
import { EventCommandStatus } from "../../enum/CommandStatus";
import { Vendor } from "../../enum/Vendor";

export interface AdapterEventDto{
    body:string;
    command:string;
    componentId:number;
    guid:string;
    isActive:boolean;
    isDefault:boolean;
    locationGuid:string;
    mac:string;
    name:string;
    reason:string;
    receivedAt?:Date;
    response:string;
    sendAt:Date;
    tag:number;
    vendor:Vendor;
    status:EventCommandStatus;
}